PassleSync.Core/ContentFinders/BaseContentFinder.cs

```diff
@@ -27,23 +30,42 @@ public bool TryFindContent(PublishedRequest request)
                 return false;
             }
 
-            if (request.Uri.Segments.Skip(1).Take(1).SingleOrDefault() != $"{UrlPrefix}/")
+            var virtualContentType = request.UmbracoContext.Content.GetContentType(ContentType);
+            var virtualContent = request.UmbracoContext.Content.GetByContentType(virtualContentType);
+            
+            if (virtualContent == null)
             {
                 return false;
             }
 
-            var shortcode = request.Uri.Segments.Reverse().Skip(1).Take(1).SingleOrDefault().Trim('/');
+            string shortcode = string.Empty;
+            if (ContentType == PassleContentType.PASSLE_AUTHOR || ContentType == PassleContentType.PASSLE_POST) 
+            {
+                // Create the regex pattern using the shortcode name and slug name
+                string regexPattern = UrlTemplate
+                       .Replace($"{{{{{ShortcodeName}}}}}", $"(?<{ShortcodeName}>.+)")
+                       .Replace($"{{{{{SlugName}}}}}", $"(?<{SlugName}>.+)");
```

Comment:
We don't need to capture the slug, so we can replace this line with a generic replace that matches all remaining variables as the only one that matters for routing is the shortcode
See this reference from the WP plugin
```php
// Replace the remaining template variables with wildcards
    // e.g. {{PassleShortcode}} will be replaced with [a-z0-9\-]+
    $regex = preg_replace("/\\\\{\\\\{[a-z0-9]+\\\\}\\\\}/i", "[a-z0-9\\-]+", $regex, -1, $count);
```